package/lib/extractPreviewRefRepositoryName.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Accepts the host string and returns the host name.
+ *
+ * @param host - Host string
+ */
+const extractFirstSubdomain = (host: string): string => host.split(".")[0];
+
+/**
+ * Parses ref object from cookie and returns proper preview link
+ *
+ * @param previewRef - Prismic Preview reference
+ */
+const extractRepositoryNameFromObjectRef = (
+	previewRef: string,
+): string | undefined => {
+	try {
+		const parsed = JSON.parse(decodeURIComponent(previewRef));
+		const keys = Object.keys(parsed);
+		const domainKey = keys.find((key) => /\.prismic\.io$/.test(key));
+
+		if (domainKey) {
+			return extractFirstSubdomain(domainKey);
+		} else {
+			return undefined;
+		}
+	} catch {
+		return undefined;
+	}
+};
+
+/**
+ * Formats preview ref to URL and returns first subdomain
```

Comment:
This description isn't completely accurate; it could be something like: "Returns the repository name from a URL-style Prismic ref."